### PR TITLE
Remove chaining from HashCodeCombiner

### PR DESCRIPTION
**Pull request by pranavkm**
Reacting to https://github.com/aspnet/Common/issues/40

### CODE REVIEW

File: submodules/FileSystem [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 8a835fc7b027418eccf1decf17206db6e12ad01a
+Subproject commit 4fbc09a477143852b6bbcd53d2cbc7e14d23d197
```
Comment (dougbu) [inline submodules/FileSystem]: ??

Comment (pranavkm) [inline submodules/FileSystem]: Updating file provider to origin/dev

**Comment by Eilon** [comment]
:shipit:

**Comment by davidfowl** [comment]
:shipit:

**Pull request closed without merging**